src/transporters/amqp10.js

```diff
@@ -0,0 +1,498 @@
+		if (!this.opts) this.opts = {};
+
+		// Number of requests a broker will handle concurrently
+		if (typeof opts.prefetch !== "number") opts.prefetch = 1;
+
+		// Number of milliseconds before an event expires
+		if (typeof opts.eventTimeToLive !== "number") opts.eventTimeToLive = null;
+
+		if (typeof opts.heartbeatTimeToLive !== "number") opts.heartbeatTimeToLive = null;
+
+		if (typeof opts.connectionOptions !== "object") opts.connectionOptions = {};
+
+		if (typeof opts.queueOptions !== "object") opts.queueOptions = {};
+
+		if (typeof opts.topicOptions !== "object") opts.topicOptions = {};
+
+		if (typeof opts.messageOptions !== "object") opts.messageOptions = {};
+
+		if (typeof opts.topicPrefix !== "string") opts.topicPrefix = "topic://";
+
+		this.receivers = [];
+		this.hasBuiltInBalancer = true;
+		this.connection = null;
+	}
+
+	_getQueueOptions(packetType, balancedQueue) {
+		let packetOptions = {};
+		switch (packetType) {
+			// Requests and responses don't expire.
+			case PACKET_REQUEST:
+				// TODO: auto delete
+				break;
+			case PACKET_RESPONSE:
+				// TODO: auto delete
+				break;
+
+			// Consumers can decide how long events live
+			// Load-balanced/grouped events
+			case PACKET_EVENT + "LB":
+			case PACKET_EVENT:
+				// TODO: auto delete
+				break;
+
+			// Packet types meant for internal use
+			case PACKET_HEARTBEAT:
+				// TODO: auto delete
+				// packetOptions = {};
+				break;
+			case PACKET_DISCOVER:
+			case PACKET_DISCONNECT:
+			case PACKET_UNKNOWN:
+			case PACKET_INFO:
+			case PACKET_PING:
+			case PACKET_PONG:
+				// TODO: auto delete
+				break;
+		}
+
+		return Object.assign(packetOptions, this.opts.queueOptions);
+	}
+
+	_getMessageOptions(packetType) {
+		let messageOptions = {};
+		switch (packetType) {
+			case PACKET_REQUEST:
+			case PACKET_RESPONSE:
+				break;
+			case PACKET_EVENT + "LB":
+			case PACKET_EVENT:
+				if (this.opts.eventTimeToLive) messageOptions.ttl = this.opts.eventTimeToLive;
+				break;
+			case PACKET_HEARTBEAT:
+				if (this.opts.heartbeatTimeToLive) messageOptions.ttl = this.opts.heartbeatTimeToLive;
+				break;
+			case PACKET_DISCOVER:
+			case PACKET_DISCONNECT:
+			case PACKET_UNKNOWN:
+			case PACKET_INFO:
+			case PACKET_PING:
+			case PACKET_PONG:
+				break;
+		}
+
+		return Object.assign(messageOptions, this.opts.messageOptions);
+	}
+
+	/**
+	 * Build a function to handle requests.
+	 *
+	 * @param {String} cmd
+	 * @param {Boolean} needAck
+	 *
+	 * @memberof Amqp10Transporter
+	 */
+	_consumeCB(cmd, needAck = false) {
+		return ({ message, delivery }) => {
+			const result = this.incomingMessage(cmd, message.body);
+
+			if (needAck) {
+				if (isPromise(result)) {
+					return result
+						.then(() => {
+							if (this.connection) {
+								delivery.accept();
+							}
+						})
+						.catch(err => {
+							this.logger.error("Message handling error.", err);
+							if (this.connection) {
+								delivery.reject();
+							}
+						});
```

Comment:
For now we can change `delivery.reject();` to 'delivery.release();' to match the 0.9.1 transporter, but FYI, it will not pass the tests, as it expects the message to be back in the queue and retryable by other nodes.

Reply:
Sorry for my slow response. Let's do `accept` in both. It might require changes in both test suites which I can help with if you'd like.